### PR TITLE
Update methods.md

### LINKED ISSUE #3549: Missing node selection methods
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/node-selection/methods

### What part(s) of the page would you like to see updated?

[These](https://github.com/dbt-labs/dbt-core/blob/39e0c22353d2230fe1bc8b4852bbb0afa27fd2c0/core/dbt/graph/selector_methods.py#L37-L55) are the methods that are currently supported.

Out of these, nearly all of them are documented. Here are the ones that are missing:
- The `resource_type` method (and example)
- The `wildcard` method (and example)
- Example that includes the `file:` method

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #3549

## What are you changing in this pull request and why?

This PR adds the (undocumented!) `resource_type` method that has been possible for a long, long time.

Reason: we discovered a selection method wasn't documented! So we did 😎 

## Not done

#3549 mentions that there are no examples for the wildcard method that include `wildcard:`. This seems okay since the intent in https://github.com/dbt-labs/dbt-core/pull/6599 was to use Unix-style wildcards in conjunction with all the other selection methods.

So we're choosing not to add `wildcard:` examples in this PR. If needed, they can always be added later.

## 🎩 

[Preview](https://deploy-preview-3548--docs-getdbt-com.netlify.app/reference/node-selection/methods#the-resource_type-method)

image

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.

### CODE REVIEW

File: website/docs/reference/node-selection/methods.md [hunk 1]
```diff
@@ -46,6 +46,18 @@ The `source` method is used to select models that select from a specified [sourc
...
+
+  ```bash
+  $ dbt build --select +resource_type:exposure    # build all resources upstream of exposures
+  $ dbt list --select +resource_type:test    # list all tests in your project
+  ```
+
+
+  ```bash
+  $ dbt run --select source:snowplow+    # run all models that select from Snowplow sources
+  ```
```
Comment (dbeatty10) [inline website/docs/reference/node-selection/methods.md]: Is this section intentional? Or is it a copy-past holdover?

Comment (dave-connors-3) [reply website/docs/reference/node-selection/methods.md]: copy paste emergency!

**Comment by dbeatty10** [comment]
@mirnawong1 would you be willing to give this one a look?

File: website/docs/reference/node-selection/methods.md [hunk 2]
```diff
@@ -46,6 +46,13 @@ The `source` method is used to select models that select from a specified [sourc
   $ dbt run --select source:snowplow+    # run all models that select from Snowplow sources
   ```
 
+### The "resource_type" method
+The `resource_type` method is used to select nodes of a particular type (`model`, `source`, `exposure`, etc). This is similar to the `--resource-type` flag used by the [`dbt ls` command](/reference/commands/list).
+
+  ```bash
+  $ dbt build --select +resource_type:exposure    # build all resources upstream of exposures
+  $ dbt list --select +resource_type:test    # list all tests in your project
```
Comment (dbeatty10) [inline website/docs/reference/node-selection/methods.md]: The ["plus" operators](https://docs.getdbt.com/reference/node-selection/graph-operators#the-plus-operator) (`+`) here seem extraneous, so I'm going to pull them out:
```suggestion
  $ dbt build --select resource_type:exposure    # build all resources upstream of exposures
  $ dbt list --select resource_type:test    # list all tests in your project
```

**Review by mirnawong1 — APPROVED** [review]
thank you @dave-connors-3 and @dbeatty10 ! non blocker sugg and lgtm 🚢!

File: website/docs/reference/node-selection/methods.md [hunk 3]
```diff
@@ -46,6 +46,13 @@ The `source` method is used to select models that select from a specified [sourc
   $ dbt run --select source:snowplow+    # run all models that select from Snowplow sources
   ```
 
+### The "resource_type" method
+The `resource_type` method is used to select nodes of a particular type (`model`, `source`, `exposure`, etc). This is similar to the `--resource-type` flag used by the [`dbt ls` command](/reference/commands/list).
```
Comment (mirnawong1) [inline website/docs/reference/node-selection/methods.md]:
```suggestion
Use the `resource_type` method to select nodes of a particular type (`model`, `source`, `exposure`, etc). This is similar to the `--resource-type` flag used by the [`dbt ls` command](/reference/commands/list).
```

**Pull request merged**